crates/aiken-project/src/blueprint/mod.rs

```diff
@@ -44,33 +48,43 @@ pub enum LookupResult<'a, T> {
     Many,
 }
 
-impl Blueprint<Schema> {
+impl Blueprint<Reference, Annotated<Schema>> {
     pub fn new(
         config: &Config,
         modules: &CheckedModules,
         generator: &mut CodeGenerator,
     ) -> Result<Self, Error> {
         let preamble = config.into();
 
+        let mut definitions = Definitions::new();
+
         let validators: Result<Vec<_>, Error> = modules
             .validators()
             .map(|(validator, def)| {
-                Validator::from_checked_module(modules, generator, validator, def)
+                Validator::from_checked_module(modules, generator, validator, def).map(
+                    |mut schema| {
+                        definitions.merge(&mut schema.definitions);
+                        schema.definitions = Definitions::new();
+                        schema
```

Comment:
Each validator is generated with its own set of definitions. When building a blueprint, all definitions are merged and put at the top-level of the blueprint.